### ISSUE TITLE
Italian translation

### ISSUE THREAD

**Issue by flates** [sepinf-inc/IPED#1091]
Dear developers,

I would like to ask to contribute to the Italian translation of the next release of IPED.
I have already completed the translation taking as base the English localized files. 
However, I noticed that in the Portuguese localized files there are a few more lines:
- "iped-desktop-messages_**.properties" -> "FileChooser.*" lines 179 to 217
- "iped-categories_**.properties" -> from line 92 onwards

I translated these parts anyway but I admit I translated the last lines of the "iped-categories__**.properties" file almost literally, missing an English reference.

I have already sent the translated files to @lfcnassif gmail account but I have no problem to fork the project and contribute via GitHub.

Kind regards,
Flavio

**Comment by lfcnassif** [comment]
Thank you very much @flatesfor contributing with this!

> the next release of IPED.

We will be able to include this in the future 4.0.0 release (within 1 month with luck), because the localization implementation was totally refactored. Possibly we will release a last 3.18.15 bug fix before that.

> * "iped-desktop-messages_**.properties" -> "FileChooser.*" lines 179 to 217

I think those were included by @tc-wleite, but not sure why since java _should_ have translation for those properties. Or it doesn't @tc-wleite? Should we include it for other languages or they are missing just in pt-BR?

> * "iped-categories_**.properties" -> from line 92 onwards

Those are custom categories related to brazilian government software that sometimes don't have similar counterparts in other countries, so the translation could be difficult... But we should translate them, since the tool will try to find related evidence and may show those categories to users outside from Brazil (maybe false positives mistakenly classified into those categories). When the translation is not found, the default english key names are used. But I will add those categories to the english localization file simply duplicating keys (key=key) since EN files are usually used as the standard for translation.

> I have already sent the translated files to @lfcnassif gmail account but I have no problem to fork the project and contribute via GitHub.

I haven't received those, could you post them here (or submit a PR)?

One last question, will you be able to help us maintain the italian translation for newly added features/modules in future versions?

Thank you

**Comment by flates** [comment]
> I haven't received those, could you post them here (or submit a PR)?

Probably the email ended up in spam. Anyway, I'll submit a PR.


> One last question, would you be able to help us maintaining the italian translation for newly added features/modules in future versions?

Absolutely, I'd be happy to!

P.S. I also translated the help and html report files :)

**Comment by lfcnassif** [comment]
> Probably the email ended up in spam

Just found it. Seems spam emails are not indexed...

> I also translated the help and html report

Great!

**Comment by tc-wleite** [comment]
> I think those were included by @tc-wleite, but not sure why since java should have translation for those properties. Or it doesn't @tc-wleite? Should we include it for other languages or they are missing just in pt-BR?

Yes, I included this because although my Locale (system and IPED) is pt_BR, JFileChooser dialogs are shown in English. 
Taking a closer look now, I noticed that the behavior changes depending on the JRE used.
I think these properties (and the code that handles them) can be removed, but there is still a minor issue: a simple test program works fine (shows everything in Portuguese) using OpenJDK 11.0.3, but it doesn't with IPED (shown in English).
I will try to find out what is going on (probably next week), and ideally remove these properties.

**Comment by lfcnassif** [comment]
> a simple test program works fine (shows everything in Portuguese) using OpenJDK 11.0.3, but it doesn't with IPED 4 (strings shown in English).

Maybe it could be related with our custom class loader or docking frames...

**Comment by lfcnassif** [comment]
@tc-wleite I saw some FileChooser localized keys being duplicated (like this https://github.com/sepinf-inc/IPED/blob/8ac0ee4be7023dc6cf3db64732bc7b718d312d88/iped-app/resources/localization/iped-desktop-messages_pt_BR.properties#L194-L195). Was that pushed by accident or do they have some purpose?

**Comment by tc-wleite** [comment]
> @tc-wleite I saw some FileChooser localized keys being duplicated like this:
> Was that pushed by accident or do they have some purpose?

It was a mistake, there shouldn't be duplicated keys.

**Comment by lfcnassif** [comment]
Ok I'll remove them. I'm replicating those FileChooser localization keys to all languages for now, until we come up with a solution to remove them.

**Comment by tc-wleite** [comment]
> Ok I'll remove them. I'm replicating those FileChooser localization keys to all languages for now, until we come up with a solution to remove them.

I am looking into a better solution...

**Comment by tc-wleite** [comment]
I think I found a possible solution. 
It should work if the used JRE has support for the language defined in IPED's locale, which I think it is the most common scenario.

It is a trivial change, can you give a try?

Change from:
```
    localizeFileChooser();
```
To:
```
    // Set the locale used by JFileChooser's
    JFileChooser.setDefaultLocale(localeConfig.getLocale());
```
 
If that works for you too, FileChooser.* could be removed from properties (and the implementation of `localizeFileChooser()` of course).

**Comment by lfcnassif** [comment]
> It is a trivial change, can you give a try?

Great, that worked! I'm going to remove them and update the code. Thank you @tc-wleite!

**Comment by lfcnassif** [comment]
Closed by #1095, thank you @flates for this contribution!
